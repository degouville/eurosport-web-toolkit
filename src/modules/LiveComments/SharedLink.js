import React from 'react';
import PropTypes from 'prop-types';
import get from 'lodash/get';
import Cards from '../../components/Card';

const SharedLink = ({ sharedlink, labelPlayButton }) => {
  const image = get(sharedlink, ['pictureurl'], '');

  const sharedLinkProps = {
    title: get(sharedlink, ['title'], ''),
    labelPlayButton: get(sharedlink, ['link', 'type'], '') === 2 ? labelPlayButton : '',
    image: image ? `${image}?w=640` : '',
    link: get(sharedlink, ['url'], ''),
  };

  return <Cards.CardSmall {...sharedLinkProps} target="_blank" />;
};

export const SharedLinkShape = PropTypes.shape({
  title: PropTypes.string,
  url: PropTypes.string,
  link: PropTypes.shape({
    type: PropTypes.number,
  }),
  pictureurl: PropTypes.string,
});

SharedLink.propTypes = {
  sharedlink: SharedLinkShape.isRequired,
  labelPlayButton: PropTypes.string.isRequired,
};

export default SharedLink;
