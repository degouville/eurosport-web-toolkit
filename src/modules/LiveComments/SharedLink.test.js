import React from 'react';
import { shallow } from 'enzyme/build';
import SharedLink from './SharedLink';
import Cards from '../../components/Card';

const props = {
  sharedlink: {
    title: 'title',
    url: 'url',
    link: {
      type: 2,
    },
    pictureurl: 'picurl',
  },
  labelPlayButton: 'play this video',
};

describe('SharedLink', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SharedLink {...props} />);
  });

  it('matches video card snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays story card if sharedlink.link type is not 2', () => {
    wrapper.setProps({
      ...props,
      sharedlink: {
        ...props.sharedlink,
        link: {
          type: 75,
        },
      },
    });

    expect(wrapper.find(Cards.CardSmall).prop('labelPlayButton')).toBe('');
  });

  it('not fails when sharedlink structure is broken', () => {
    wrapper.setProps({
      sharedlink: {},
    });

    expect(wrapper.find(Cards.CardSmall).props()).toEqual({
      description: '',
      image: '',
      labelPlayButton: '',
      link: '',
      title: '',
      topic: '',
      target: '_blank',
      darkness: 0,
    });
  });
});
