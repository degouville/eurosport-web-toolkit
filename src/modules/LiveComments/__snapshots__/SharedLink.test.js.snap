// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedLink matches video card snapshot 1`] = `
<Styled(Component)
  darkness={0}
  description=""
  image="picurl?w=640"
  labelPlayButton="play this video"
  link="url"
  target="_blank"
  title="title"
  topic=""
/>
`;
